src/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -125,5 +114,64 @@ public static IEnumerable<string> GetAllProjectFileNames(string solutionFile)
                 throw new CommandLineException(message);
             }
         }
+
+        public static string GetMsbuildDirectory(string version)
+        {
+            if (String.IsNullOrEmpty(version))
+            {
+                // default to 4.0
+                version = "4.0";
+
+                // run msbuild to get the version
+                var processStartInfo = new ProcessStartInfo
+                {
+                    UseShellExecute = false,
+                    FileName = "msbuild.exe",
+                    Arguments = "/version",
+                    RedirectStandardOutput = true
+                };
+                try
+                {
+                    using (var process = Process.Start(processStartInfo))
+                    {
+                        process.WaitForExit(100);
```

Comment:
Don't know what's the right value to use here. We are just running `msbuild /version` so 100ms seems quite enough to me.

But I don't really like this auto detection of msbuild since this makes the behavior kind of indeterministic. I'll check with @yishaigalatzer to see if we can get rid of this.